core/kazoo/src/kz_doc.erl

```diff
@@ -205,14 +211,28 @@ add_pvt_document_hash(JObj, _, _) ->
 public_fields(Thing) ->
     public_fields(Thing, 'true').
 
--spec public_fields(kz_json:object() | kz_json:objects(), boolean()) ->
+-spec public_fields(kz_json:object() | kz_json:objects(), boolean() | private_fields()) ->
                            kz_json:object() | kz_json:objects().
 public_fields(JObjs, IncludeId) when is_list(JObjs) ->
     [public_fields(J, IncludeId) || J <- JObjs];
 public_fields(JObj, 'true') ->
     kz_json:set_value(<<"id">>, id(JObj, 'null'), filter_public_fields(JObj));
 public_fields(JObj, 'false') ->
-    filter_public_fields(JObj).
+    filter_public_fields(JObj);
+public_fields(JObj, Leaks) ->
```

Comment:
Hmm, shouldn't these leak into a `_read_only` object? So `kz_json:set_value([<<"_read_only">>, Key], ...)` below?

Reply:
we're already doing it for `id`. seemed to me a good way to leak a list of private fields.